[messages] database clean up

KrikTze opened the pull request:
[Messages] - Create a background job that periodically cleans up the database

KrikTze committed:
Create the initial queries for database cleanup

KrikTze committed:
Make some improvements on the delete query

KrikTze committed:
Add database clean up service
Add tests
Try to configure it to work as a background service

KrikTze committed:
Merge branch 'develop' into feature/messages/database-cleanup

KrikTze committed:
Create mock fileservice for tests - change catch

KrikTze committed:
Add CronExpression inside the options

KrikTze committed:
Clean up the service

KrikTze committed:
Merge branch 'develop' into feature/messages/database-cleanup

KrikTze committed:
Remove Clean up service from the campaign Handler

travlos committed:
Merge branch 'develop' into feature/messages/database-cleanup

travlos commented on src/Indice.Features.Messages.Core/Models/CampaignDLId.cs:
```diff
@@ -0,0 +1,22 @@
+/// <summary>
+/// An intermediate object that captures a campaign Id and its distribution list Id.
+/// </summary>
+public class CampaignDLId
```
Convert it to a record

travlos commented on src/Indice.Features.Messages.Core/Handlers/DatabaseCleanUpHandler.cs:
```diff
@@ -0,0 +1,30 @@
+/// </summary>
+public sealed class DatabaseCleanUpHandler : ICampaignJobHandler<DatabaseCleanUpTimerEvent>
+{
+    private readonly DatabaseCleanUpService _cleanUpService;
```
Inject the interface

travlos commented on src/Indice.Features.Messages.Core/DatabaseCleanUpOptions.cs:
```diff
@@ -0,0 +1,27 @@
+    /// <summary>
+    /// Gets or sets the retention period for campaigns with inbox (in days).
+    /// </summary>
+    public int CampaignsWithInboxRetentionPeriodInDays { get; set; } = 120;
```
Rename to RetentionDaysForInbox

travlos commented on src/Indice.Features.Messages.Core/DatabaseCleanUpOptions.cs:
```diff
@@ -0,0 +1,27 @@
+    /// <summary>
+    /// Gets or sets the retention period for campaigns without inbox (in days).
+    /// </summary>
+    public int CampaignsWithoutInboxRetentionPeriodInDays { get; set; } = 30;
```
Rename to RetentionDaysForOther

travlos commented on src/Indice.Features.Messages.Core/DatabaseCleanUpOptions.cs:
```diff
@@ -0,0 +1,27 @@
+/// <summary>
+/// Options for database cleanup operations - retention policies.
+/// </summary>
+public class DatabaseCleanUpOptions
```
Add a flag to turn off this Feature on/off

travlos commented on src/Indice.Features.Messages.Worker.Azure/HostBuilderExtensions.cs:
```diff
@@ -98,6 +103,7 @@ private static IServiceCollection AddCoreServices(this IServiceCollection servic
         services.TryAddTransient<CreateCampaignRequestValidator>();
         services.TryAddTransient<CreateMessageTypeRequestValidator>();
         services.TryAddTransient<NotificationsManager>();
+        services.TryAddTransient<DatabaseCleanUpService>();
```
Add Interface

KrikTze committed:
refactor certain things based on the PR

KrikTze committed:
Merge branch 'develop' into feature/messages/database-cleanup

KrikTze committed:
Merge branch 'develop' into feature/messages/database-cleanup

KrikTze committed:
Update retention policy

KrikTze committed:
Fix based on copilot suggestions

travlos committed:
Refactor attachment path logic & improve cleanup docs

Centralize file path construction for campaign attachments via DbAttachment.GetPath(), replacing inline logic throughout the codebase. Enhance XML documentation for database cleanup options, events, and scheduling, clarifying defaults and intent. Remove unused usings and improve code comments for maintainability.

travlos committed:
Remove unused BouncyCastle using in DbAttachment.cs

Removed an unnecessary using directive for Org.BouncyCastle.Bcpg from DbAttachment.cs to clean up the code and eliminate unused dependencies.

travlos committed:
Refactor DatabaseCleanUp function signature

Remove DatabaseCleanUpHandler parameter from the DatabaseCleanUp Azure Function. The function now only accepts TimerInfo and creates the handler internally using CleanUpJobHandlerFactory, simplifying the signature and centralizing handler instantiation. Also updated XML documentation accordingly.

travlos committed:
Move DatabaseCleanUpOptions to MessageWorkerOptions

DatabaseCleanUpOptions is now defined in MessageWorkerOptions instead of MessageOptions, making cleanup configuration available at the worker level. Also, DatabaseCleanUpOptions is now explicitly configured in DI, and redundant property definitions have been removed to clarify the configuration hierarchy.

travlos committed:
Improve logging and error handling in cleanup service

- Add ILogger to DatabaseCleanUpService for better logging.
- Log exceptions during campaign and attachment deletions.
- Update Messages version prefix to .2.
- Enhances observability and troubleshooting of cleanup tasks.
